packages/polling-controller/src/PollingController.ts

```diff
@@ -1,11 +1,25 @@
 import { BaseController, BaseControllerV2 } from '@metamask/base-controller';
 import type { NetworkClientId } from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+import stringify from 'fast-json-stable-stringify';
 import { v4 as random } from 'uuid';
 
 // Mixin classes require a constructor with an `...any[]` parameter
 // See TS2545
 type Constructor = new (...args: any[]) => object;
 
+/**
+ * Returns a unique key for a networkClientId and options. This is used to group networkClientId polls with the same options
+ * @param networkClientId - The networkClientId to get a key for
+ * @param options - The options used to group the polling events
+ * @returns The unique key
+ */
+export const getKey = (
+  networkClientId: NetworkClientId,
+  options: Json,
+): PollingGroupId => `${networkClientId}:${stringify(options)}`;
+
+type PollingGroupId = `${NetworkClientId}:${string}`;
```

Comment:
[nit] last thing is maybe we should align with the `pollingTokenSet` verbiage -> `pollingTokenSetId`?